std/container/rbtree.d

```diff
@@ -1806,6 +1849,41 @@ auto redBlackTree(alias less, bool allowDuplicates, E)(E[] elems...)
     return new RedBlackTree!(E, binaryFun!less, allowDuplicates)(elems);
 }
 
+
+import std.range.primitives: isInputRange, isImplicitlyConvertible, ElementType;
+
+/++ Ditto +/
+auto redBlackTree(Stuff)(Stuff range)
+if (isInputRange!Stuff && !is(typeof(range) == string))
+{
+    return new RedBlackTree!(ElementType!Stuff)(range);
+}
+
+/++ Ditto +/
+auto redBlackTree(bool allowDuplicates, Stuff)(Stuff range)
+if (isInputRange!Stuff && !is(typeof(range) == string))
+{
+    return new RedBlackTree!(ElementType!Stuff, "a < b", allowDuplicates)(range);
+}
+
+/++ Ditto +/
+auto redBlackTree(alias less, Stuff)(Stuff range)
+if (isInputRange!Stuff && !is(typeof(range) == string))
+{
+    return new RedBlackTree!(ElementType!Stuff, less)(range);
+}
+
+/++ Ditto +/
+auto redBlackTree(alias less, bool allowDuplicates, Stuff)(Stuff range)
+    if(is(typeof(binaryFun!less(E.init, E.init))) && isInputRange!Stuff &&
+       !is(typeof(range) == string))
+{
+    //We shouldn't need to instantiate less here, but for some reason,
+    //dmd can't handle it if we don't (even though the template which
+    //takes less but not allowDuplicates works just fine).
+    return new RedBlackTree!(ElementType!Stuff, binaryFun!less, allowDuplicates)(range);
+}
+
```

Comment:
When I initially fixed the constructor to support range construction I intentionally did not add these because the `redBlackTree` overload set is terribly designed. With this, there are 8 overloads of `redBlackTree` just to accomodate arbitrary argument order, one of which is a classical case of boolean blindness. Blergh.